src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -757,6 +788,9 @@ impl<'a, 'tcx> Checker<'a, 'tcx> {
                         // `let _: &'static _ = &(Cell::new(1), 2).1;`
                         let mut local_qualifs = self.qualifs_in_local(local);
                         local_qualifs[HasMutInterior] = false;
+                        // Make sure there is no reason to prevent promotion.
+                        // This is, in particular, the "implicit promotion" version of
+                        // the check making sure that we don't run drop glue during const-eval.
```

Comment:
This seems oddly specific - this *is* the original sole check, so comparing it to something newer seems off.
You could move the comment one line above and make it more about "any qualifications, except `HasMutInterior` (see above), disqualify from promotion".

Reply:
I don't think which check came first historically has any bearing here. Just because it grw that way doesn't mean that's the best way to explain it.

The other check is IMO clearly "more principled" by looking at the place where the side-effect actually occurs.

Reply:
I have adjusted the wording a bit.